### ISSUE TITLE
migrating to 6.1.1 -> 6.5+: exception occurs applying java-platform

### ISSUE THREAD

**Issue by rpattabi** [gradle/gradle#14832]
I have BoM which I use within my app projects. This worked fine with gradle 6.1.1. After I update gradle to 6.6.1, I get this error during gradle sync:

```
An exception occurred applying plugin request [id: 'java-platform']
> Failed to apply plugin 'org.gradle.java-platform'.
   > Project#afterEvaluate(Action) on project ':bom' cannot be executed in the current context.
```

Full exception stack is [here](https://gist.github.com/rpattabi/f099c12d82f5ff840d74f4089bff79e2).

I wonder if I'm missing anything. I'm not an gradle expert. I reviewed [migration documentation](https://docs.gradle.org/current/userguide/upgrading_version_6.html#changes_6.1). There was a mention about default category being library. But I don't know if it is related to my problem or what to do about that.

### Expected Behavior

BoM should build just like it did in 6.1.1

### Current Behavior

Exception during gradle sync. Same error for 6.5 through 6.6.1.

### Context

This BoM is part of android project. I had to upgrade gradle from 6.1.1 as the new android gradle plugin 4.1 is dependent on gradle 6.5+.

My BoM has the following code as mentioned in the [documentation](https://docs.gradle.org/current/userguide/java_platform_plugin.html):

```
plugins {
    id 'java-platform'
}

group rootProject.ext.group
version versions.Bom

javaPlatform {
    allowDependencies()
}

dependencies {
    api platform("$deps.firebase.bom:$bom.firebaseBom")

    constraints {
        api "$deps.multidex:$bom.multidex"

// etc
```

### Your Environment

Ubuntu 18.04
Android Studio 4.1
Gradle 6.6.1

**Comment by melix** [comment]
Would you mind sharing details about the flags that you are using? I can see that you have "configure on demand" enabled for example.

**Comment by rpattabi** [comment]
The following is what I have in my `gradle.properties`. Mostly android related flags.

```
android.enableJetifier=true
android.useAndroidX=true
android.debug.obsoleteApi=true
android.bundle.enableUncompressedNativeLibs=false

org.gradle.jvmargs=-Xms128m -Xmx1024m -XX:+CMSClassUnloadingEnabled
```

I'm using Android Studio. So I presume that it brings in its own flags, one of which may be "configure on demand".

PS: I'm currently not creating BoM in my project, thus avoiding this problem. I was using BoM in order to share dependencies across different android studio projects. I found combining those modules to a single project made things simpler avoiding the need for BoM.

**Comment by ljacomet** [comment]
Any chance you can provide a reproducer? Without one, this will be hard to follow upon.

**Comment by rpattabi** [comment]
As mentioned, I stopped using BoM which turned out to be right approach for my project. It would be difficult to provide a reproducer at this time if the provided info is not resulting in reproducer.

It's been about 2 months and it looks like no-one seems to have faced this problem except me. So I propose to close this issue.

**Comment by ljacomet** [comment]
Ok, thanks for following up.

Closing since we could not reproduce either.
